File: database/database-annotation-processor/src/test/java/io/koraframework/database/common/annotation/processor/repository/error/QuotedQueryPlaceholder.java
package io.koraframework.database.common.annotation.processor.repository.error;

import io.koraframework.database.common.annotation.Query;
import io.koraframework.database.common.annotation.Repository;
import io.koraframework.database.jdbc.JdbcRepository;

@Repository
public interface QuotedQueryPlaceholder extends JdbcRepository {

    @Query("""
        SELECT
          ':missing',
          'it''s :escaped',
          ":alsoMissing",
          $$:dollarMissing$$,
          $tag$:tagMissing$tag$,
          `:backtickMissing`,
          [:bracketMissing],
          *
        FROM table
        -- :lineCommentMissing
        WHERE id = :id
        /* :blockCommentMissing */
        """)
    String quotedPlaceholderIsIgnored(long id);
}

File: database/database-symbol-processor/src/test/kotlin/io/koraframework/database/symbol/processor/repository/error/QuotedQueryPlaceholder.kt
package io.koraframework.database.symbol.processor.repository.error

import io.koraframework.database.common.annotation.Query
import io.koraframework.database.common.annotation.Repository
import io.koraframework.database.jdbc.JdbcRepository

@Repository
interface QuotedQueryPlaceholder : JdbcRepository {

    @Query("SELECT ':missing', 'it''s :escaped', \":alsoMissing\", \$\$:dollarMissing\$\$, \$tag\$:tagMissing\$tag\$, `:backtickMissing`, [:bracketMissing], * FROM table -- :lineCommentMissing\n WHERE id = :id /* :blockCommentMissing */")
    fun quotedPlaceholderIsIgnored(id: Long): String
}
